criu/config.c

```diff
@@ -46,53 +46,89 @@ int parse_statement(int i, char *line, char **configuration)
 {
 	int offset = 0, len = 0;
 	bool was_newline = true;
-	char *tmp_string, *quoted, *quotedptr;
+	char *tmp_string;
 
 	while (1) {
 		/* Ignore white-space */
 		while ((isspace(*(line + offset)) && (*(line + offset) != '\n'))) offset++;
 
+		/* Ignore line starting with a comment */
+		if (line[offset] == '#')
+			break;
+
 		/* Read a single word. A word is everything
 		 * that doesn't contain white-space characters. */
 		if (sscanf(line + offset, "%m[^ \t\n]s", &configuration[i]) != 1) {
 			configuration[i] = NULL;
 			break;
 		}
 
-		/* Ignore comments - everything between '#' and '\n' */
-		if (configuration[i][0] == '#') {
-			configuration[i] = NULL;
-			break;
+		if (configuration[i][0] == '"') {
+			int quote_offset;
+			char *quote_start;
+
+			/* Move by one to skip the leading quote. */
+			memmove(configuration[i], configuration[i] + 1, len - 1);
+			offset++;
+			quote_start = line + offset;
+			quote_offset = offset;
+
+			/*
+			 * If it starts with a quote everything until the
+			 * next unescaped quote needs to be looked at.
+			 */
+			while ((tmp_string = strchr(line + quote_offset + 1, '"'))) {
+				quote_offset += tmp_string - (line + quote_offset);
+				/* Check if it is escaped */
+				if (*(tmp_string - 1) == '\\')
+					continue;
+
+				/* Not escaped. That is the end of the quoted string. */
+				configuration[i] = xrealloc(configuration[i], quote_offset - offset + 1);
```

Comment:
coverity will not like this. It will tell you that here can be a memory leak.

new = xrealloc(configuration[i]configuration[i], quote_offset - offset + 1);
if (new == NULL) {
  ... fail
}
configuration[i] = new;